### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by dimakrugly**
[DEMO LINK](https://dimakrugly.github.io/react_redux-list-of-todos/)

### CODE REVIEW

File: src/store/reducers/todos.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  error: string | null;
+}
+
+const initialState: TodosState = {
+  todos: [],
+  loading: false,
+  error: null,
+};
+
+const todosReducer = (state = initialState, action: any): TodosState => {
```
Comment (vadiimvooo) [inline src/store/reducers/todos.ts:20]: Please use action type here. It's bed approach to use any.

File: src/store/reducers/user.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+type UserRemoveAction = {
+  type: 'USER_REMOVE';
+};
+
+export const removeUser = (): UserRemoveAction => ({
+  type: 'USER_REMOVE',
+});
+
+const userReducer = (state = initialState, action: any): UserState => {
```
Comment (vadiimvooo) [inline src/store/reducers/user.ts:29]: The same is here.

File: src/store/actionCreators/user.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  type: 'FETCH_USER_SUCCESS';
+  payload: User;
+};
+
+type FetchUserFailureAction = {
+  type: 'FETCH_USER_FAILURE';
+  payload: string;
+};
+
+/* eslint-disable-next-line */
```
Comment (vadiimvooo) [inline src/store/actionCreators/user.ts:25]: Remove this rule and just separate on different line the code.

File: src/store/actionCreators/user.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  type: FETCH_USER_SUCCESS,
+  payload: user,
+});
+
+export const fetchUserFailure = (error: string): FetchUserFailureAction => ({
+  type: FETCH_USER_FAILURE,
+  payload: error,
+});
+
+/* eslint-disable-next-line */
```
Comment (vadiimvooo) [inline src/store/actionCreators/user.ts:42]: Remove this rule and just separate on different line the code.

File: src/store/actionCreators/todos.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  type: 'FETCH_TODOS_SUCCESS';
+  payload: Todo[];
+};
+
+type FetchTodosFailureAction = {
+  type: 'FETCH_TODOS_FAILURE';
+  payload: string;
+};
+
+/* eslint-disable-next-line */
```
Comment (vadiimvooo) [inline src/store/actionCreators/todos.ts:25]: Remove this rule and just separate on different line the code.

File: src/store/actionCreators/todos.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+export const fetchTodosAsync = (): ThunkAction<void, RootState, unknown, Action> =>
+  async dispatch => {
+    try {
+      dispatch(fetchTodosRequest());
+      const data = await getTodos();
+
+      dispatch(fetchTodosSuccess(data));
+    } catch (error: any) {
```
Comment (vadiimvooo) [inline src/store/actionCreators/todos.ts:50]: It's possible to use here any, but add eslint-disable-next-line

File: src/store/actionCreators/todos.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  type: FETCH_TODOS_SUCCESS,
+  payload: todos,
+});
+
+export const fetchTodosFailure = (error: string): FetchTodosFailureAction => ({
+  type: FETCH_TODOS_FAILURE,
+  payload: error,
+});
+
+/* eslint-disable-next-line */
```
Comment (vadiimvooo) [inline src/store/actionCreators/todos.ts:42]: Remove this rule and just separate on different line the code.

File: src/components/TodoList/TodoList.tsx [hunk 8]
```diff
@@ -1,13 +1,30 @@
...
+  currentTodoId: number;
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  onAddCurrentTodo,
+  currentTodoId,
+}) => {
+
+  if (todos.length === 0) {
```
Comment (vadiimvooo) [inline src/components/TodoList/TodoList.tsx:18]:
```suggestion
  if (!todos.length) {
```

File: src/components/TodoList/TodoList.tsx [hunk 9]
```diff
@@ -25,198 +42,38 @@ export const TodoList: React.FC = () => {
...
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">4</td>
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
+          {todos.map(todo => (
+            <tr data-cy="todo" key={todo.id}>
```
Comment (vadiimvooo) [inline src/components/TodoList/TodoList.tsx:46]: This table row element can be separated from TodoList as component TodoItem.

File: src/components/TodoList/TodoList.tsx [hunk 10]
```diff
@@ -25,198 +42,38 @@ export const TodoList: React.FC = () => {
...
+
+              <td className="has-text-right is-vcentered">
+                <button
+                  data-cy="selectButton"
+                  className="button"
+                  type="button"
+                  onClick={()=>onAddCurrentTodo(todo)}
+                >
+                  <span className="icon">
+                    <i className={`far ${currentTodoId === todo.id ? 'fa-eye-slash' : 'fa-eye'}`} />
```
Comment (vadiimvooo) [inline src/components/TodoList/TodoList.tsx:71]: Use classnames lib for className

**Review by vadiimvooo — CHANGES_REQUESTED** [review]
Well done. There is some changes needs to be done. Also try to fix linter errors.

File: src/store/actionCreators/user.ts (hunk 4 above)
Comment (dimakrugly) [reply src/store/actionCreators/user.ts:42]: impossible, i can not do it

File: src/components/TodoList/TodoList.tsx (hunk 10 above)
Comment (dimakrugly) [reply src/components/TodoList/TodoList.tsx:71]: it was much more readable when it was without cn))

const iconClass = classNames('far', {
  'fa-eye-slash': currentTodoId === todo.id,
  'fa-eye': currentTodoId !== todo.id,
});

**Comment by dimakrugly** [comment]
> Well done. There is some changes needs to be done. Also try to fix linter errors.

**Pull request closed without merging**